src/libsync/theme.cpp

```diff
@@ -280,6 +280,85 @@ QIcon Theme::uiThemeIcon(const QString &iconName, bool uiHasDarkBg) const
 }
 
+QPixmap Theme::pixmapForBackground(const QString &fileName, const QColor &backgroundColor) const
+{
+    Q_ASSERT(!fileName.isEmpty());
+
+    const auto iconBasePath = QStringLiteral(":/client/theme");
+
+    // some icons are present in white or black only, so, we need to check both when needed
+    const auto iconBaseColors = QStringList({ QStringLiteral("black"), QStringLiteral("white") });
+
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+
+    const QString cacheKey = fileName + QLatin1Char(',') + pixmapColor;
+
+    auto &cachedPixmap = _pixmapCache[cacheKey];
+
+    if (cachedPixmap.isNull()) {
+        if (iconBaseColors.contains(pixmapColor)) {
+            return cachedPixmap = QPixmap::fromImage(QImage(iconBasePath + QLatin1Char('/') + pixmapColor + QLatin1Char('/') + fileName));
+        }
+
+        const auto drawSvgWithCustomFillColor = [](const QString &sourceSvgPath, const QString &fillColor) {
+            QSvgRenderer svgRenderer;
+            
+            if (!svgRenderer.load(sourceSvgPath)) {
+                return QPixmap();
+            }
+
+            // render source image
+            QImage svgImage(svgRenderer.defaultSize(), QImage::Format_ARGB32);
+            {
+                QPainter svgImagePainter(&svgImage);
+                svgImage.fill(Qt::GlobalColor::transparent);
+                svgRenderer.render(&svgImagePainter);
+            }
+
+            // draw target image with custom fillColor
+            QImage image(svgRenderer.defaultSize(), QImage::Format_ARGB32);
+            image.fill(QColor(fillColor));
+            {
+                QPainter imagePainter(&image);
+                imagePainter.setCompositionMode(QPainter::CompositionMode_DestinationIn);
+                imagePainter.drawImage(0, 0, svgImage);
+            }
+
+            return QPixmap::fromImage(image);
+        };
+
+        // find the first matching svg among base colors, if any
+        const QString sourceSvg = [&]() {
+            for (const auto &color : iconBaseColors) {
+                const QString baseSVG(iconBasePath + QLatin1Char('/') + color + QLatin1Char('/') + fileName);
+
+                if (QFile(baseSVG).exists()) {
+                    return baseSVG;
+                }
+            }
+            return QString();
+        }();
+
+        Q_ASSERT(!iconBaseColors.isEmpty());
+        if (iconBaseColors.isEmpty()) {
+            qWarning("Failed to find base svg for %s", qPrintable(cacheKey));
+            _pixmapCache.remove(cacheKey);
+            return {};
+        }
+
+        cachedPixmap = drawSvgWithCustomFillColor(sourceSvg, pixmapColor);
+
+        Q_ASSERT(!cachedPixmap.isNull());
+        if (cachedPixmap.isNull()) {
+            qWarning("Failed to load pixmap for %s", qPrintable(cacheKey));
+            _pixmapCache.remove(cacheKey);
+            return {};
+        }
+    }
+
+    return cachedPixmap;
+}
```

Comment:
Maybe you could extract that pixmap cache logic into its own class. Then it will also be possible to add unit tests to it. At least the method could be made a bit more readable/smaller if you can extract the pixmap cache logic in it's own function like `pixmapForBackgroundFromCache()`.

Reply:
@FlexW Hmm, I am not sure if it makes sense to add new files (for a new class) when it is just one method really. I can add unit tests to this method in fact.

Reply:
Okay :) I just like it if a function does one thing. This function seems to do a lot of things.

Reply:
I think that this feature will be useful as a general feature (and we may in the future need to change the internal implementation in case of icons with two colors)
Please follow the comment from @FlexW

Reply:
@matthieugallien But how will it be possible to use 2 colors without CSS? Won't this require a completely different approach? Also, by design, we are unlikely going to have colorful icons. But, this is fine, I can change something just to make this PR through.

Reply:
for now your solution is enough
I am fine with it